modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonadLaws.kt

```diff
@@ -76,16 +76,16 @@ object MonadLaws {
   fun <F> Monad<F>.equivalentComprehensions(EQ: Eq<Kind<F, Int>>) {
     val M = this
     forAll(Gen.int()) { num: Int ->
-      val aa = binding {
-        val a = just(num).bind()
-        val b = just(a + 1).bind()
-        val c = just(b + 1).bind()
+      val aa = fx {
+        val (a) = just(num)
```

Comment:
yes, except `effect` will take a `suspend () -> A` and bind directly to `A`. 
`effect` is just a compile time trick and can be inlined to the `effectIdentity`. This is how it's implemented:
```kotlin
suspend fun <A> effect(f: suspend () -> A): A = f()
```
It does nothing but force the user to use it whenever a side effect is denoted to be injested from the environment into the continuation. At the moment this is only available for MonadDefer and up because those are the only ones that can suspend side effects.